include/aspect/simulator.h

```diff
@@ -968,6 +968,44 @@ namespace aspect
       get_extrapolated_advection_field_range (const AdvectionField &advection_field) const;
 
       /**
+       * Check if an additional refinement cycle is required in this timestep.
+       */
+      bool time_for_additional_refinement_cycle () const;
```

Comment:
I think we've been pretty good at annotating each function's documentation with the file in which it is implemented.